library/src/scala/tasty/reflect/SymbolOps.scala

```diff
@@ -83,181 +63,63 @@ trait SymbolOps extends Core {
     def isAbstractType(given ctx: Context): Boolean = internal.Symbol_isAbstractType(self)
     def isClassConstructor(given ctx: Context): Boolean = internal.Symbol_isClassConstructor(self)
 
-    def isType(given ctx: Context): Boolean = internal.matchTypeSymbol(self).isDefined
-    def isTerm(given ctx: Context): Boolean = internal.matchTermSymbol(self).isDefined
-    def isValDef(given ctx: Context): Boolean = internal.matchValDefSymbol(self).isDefined
-    def isDefDef(given ctx: Context): Boolean = internal.matchDefDefSymbol(self).isDefined
-    def isClass(given ctx: Context): Boolean = internal.matchClassDefSymbol(self).isDefined
-  }
-
-  // PackageSymbol
-
-  object IsPackageDefSymbol {
-    def unapply(symbol: Symbol)(given ctx: Context): Option[PackageDefSymbol] =
-      internal.matchPackageDefSymbol(symbol)
-  }
-
-  implicit class PackageDefSymbolAPI(self: PackageDefSymbol) {
-    def tree(given ctx: Context): PackageDef =
-      internal.PackageDefSymbol_tree(self)
-  }
-
-  // TypeSymbol
-
-  object IsTypeSymbol {
-    def unapply(symbol: Symbol)(given ctx: Context): Option[TypeSymbol] =
-      internal.matchTypeSymbol(symbol)
-  }
-
-  // ClassSymbol
-
-  object IsClassDefSymbol {
-    def unapply(symbol: Symbol)(given ctx: Context): Option[ClassDefSymbol] =
-      internal.matchClassDefSymbol(symbol)
-  }
-
-  object ClassDefSymbol {
-    /** The ClassSymbol of a global class definition */
-    def of(fullName: String)(given ctx: Context): ClassDefSymbol =
-      internal.ClassDefSymbol_of(fullName)
-  }
-
-  implicit class ClassDefSymbolAPI(self: ClassDefSymbol) {
-    /** ClassDef tree of this defintion */
-    def tree(given ctx: Context): ClassDef =
-      internal.ClassDefSymbol_tree(self)
+    def isType(given ctx: Context): Boolean = internal.Symbol_isTypeSymbol(self)
+    def isTerm(given ctx: Context): Boolean = internal.Symbol_isTermSymbol(self)
+    def isValDef(given ctx: Context): Boolean = internal.Symbol_isValDefSymbol(self)
+    def isDefDef(given ctx: Context): Boolean = internal.Symbol_isDefDefSymbol(self)
+    def isTypeDef(given ctx: Context): Boolean = self.isType && !self.flags.is(Flags.Case)
+    def isClassDef(given ctx: Context): Boolean = internal.Symbol_isClassDefSymbol(self)
+    def isBind(given ctx: Context): Boolean = self.isTerm && self.flags.is(Flags.Case)
+    def isTypeBind(given ctx: Context): Boolean = self.isType && self.flags.is(Flags.Case)
+    def isPackageDef(given ctx: Context): Boolean = internal.Symbol_isPackageDefSymbol(self)
+    def isNoSymbol(given ctx: Context): Boolean = self == Symbol.noSymbol
```

Comment:
`isDefDef ` -> `isMethod`?
`isTypeDef` -> `isType`?
`isClassDef` -> `isClass`?
`isBind` -> `isPatternBoundTerm`?
`isTypeBind` -> `isPatternBoundType`?
`isPackageDef` -> `isPackage`?
`isNoSymbol` -> `exists`?

Reply:
We already have an `isType. Also, those are supposed to have the names of the trees they represent.

Reply:
Thank for clarification, then it's better to leave as it is.

Reply:
I added documentation to show clearly the relation between the symbols and the trees.